### PR TITLE
Settings: Improve titles

### PR DESCRIPTION
**Pull request by dhruvisompura**
### Description
Addressess #1487

This is a first pass at improving the setting titles displayed in the settings table of contents for built in positron extensions. Any titles prefixed with "Positron" has been removed and ambiguity in extension titles have been addressed. The setting title in the TOC and category title for each setting item has been unified to match where possible. configuration keys have been changed to fix formatting of setting categories where possible.

Extension configuration keys that are prefixed with `positron` have been left as is for now (excluding theInterpreter settings) since the prefix does not effect the settings UI for now. We may want to address improving the key names for folks who interact with settings via the JSON file. It may be worth tackling after settling on a standard for configuration name formatting for bundled extensions.

The "Positron > Interpreters" settings now live under "Interpreters"/

The following extensions were modified and have the following setting titles in the TOC:
- open-remote-ssh: Remote SSH
- positron-notebook-controllers: Notebook Controllers
- positron-r: R
- positron-reticulate: Reticulate
- positron-run-app: App Launcher
- positron-supervisor: Kernel Supervisor
- positron-viewer: URL Viewer

### QA Notes

We'll want to verify there aren't any regressions with the settings for the extensions listed above, excluding any experimental feature settings.

For the interpreter settings we should verify that changing the automatic startup behavior, restart on crash behavior, and viewing sessions for interpreters are respected as usual. Some of this behavior should be captured by our e2e tests which will be run against this change before merging.

Note that as part of this change, the qa-example-content repo did need a change due to a setting key being renamed: https://github.com/posit-dev/qa-example-content/pull/41

### Screenshots

https://github.com/user-attachments/assets/fc66391a-5e0e-49a9-b717-a0d5401e2d1e

### CODE REVIEW

**Comment by dhruvisompura** [comment]
To make it easier to review this PR, I would recommend hiding whitespace to only see relevant changes to updating setting titles.

File: extensions/positron-r/package.nls.json
Comment (dhruvisompura) [inline extensions/positron-r/package.nls.json:34]: I am considering changing this title to: `Advanced` since the `Settings` portion is a bit redundant. Let me know if anyone strongly disagrees. The contribution point docs mention avoiding these terms since they are redundant as well: https://code.visualstudio.com/api/references/contribution-points#contributes.configuration

Comment (juliasilge) [reply extensions/positron-r/package.nls.json:34]: Yes, let's _not_ use "Settings" 👍

**Review by juliasilge — DISMISSED** [review]
This looks fantastic! The state of our settings has been bothering me for soooooo long. 😭 

I especially am happy about the rename to "App Launcher". Can you open up an issue to update [these docs](https://positron.posit.co/run-interactive-apps)?

Also, since this is such an extensive set of changes, can you partner with QA to make sure we are running an extremely comprehensive set of tests on this PR and that we are all green before merging?

File: extensions/positron-run-app/package.json
Comment (dhruvisompura) [inline extensions/positron-run-app/package.json:19]: I noticed that we have quite a number of settings that live under the `positron` prefix in the JSON file for settings. It's a little odd from a UX perspective for some setting IDs to have a `positron` prefix. From a DX standpoint, its helpful but we may want to consider removing this prefix and adding some guidelines to the wiki on naming these properties.

What do others think?

Comment (juliasilge) [reply extensions/positron-run-app/package.json:19]: I do not have strong opinions on this, except that I don't think it is nearly as important as what folks see in the UI. Maybe not urgent enough to do right now, and instead we can open an issue to reexamine later?

I do think it's possible this would be a difficult change to make because many, many beta users have these settings in their `settings.json` already with their original names.

**Review by juliasilge — APPROVED** [review]
LGTM, pending some very thorough test runs! 🚀

File: extensions/positron-run-app/package.json
Comment (dhruvisompura) [reply extensions/positron-run-app/package.json:19]: > I do not have strong opinions on this, except that I don't think it is nearly as important as what folks see in the UI. Maybe not urgent enough to do right now, and instead we can open an issue to reexamine later?

I agree that it doesn't feel urgent. I'll drop a note about this in the tagged issue for this PR since it's all settings related. 

> I do think it's possible this would be a difficult change to make because many, many beta users have these settings in their settings.json already with their original names.

Totally agree that it is a difficult change but I would argue that its better to make this change while we are in beta. It would be more acceptable before an official release of the product. It's definitely lower priority and maybe not worth the effort to change if it will cause disruption for users without much gains. I was just curious if other folks agreed/disagreed.

**Comment by dhruvisompura** [comment]
> This looks fantastic! The state of our settings has been bothering me for soooooo long. 😭
> 
> I especially am happy about the rename to "App Launcher". Can you open up an issue to update [these docs](https://positron.posit.co/run-interactive-apps)?

Created an issue to update the docs page: https://github.com/posit-dev/positron/issues/5824

> Also, since this is such an extensive set of changes, can you partner with QA to make sure we are running an extremely comprehensive set of tests on this PR and that we are all green before merging?

I chatted with QA and we'll run the full test suite on this to make sure none of the settings were broken in the process of the rename. We don't have setting specific tests but there are a number of tests that update settings which should cover the changes.

**Pull request merged**
